client/src/components/Tea/TeaList/TeaListContainer.jsx

```diff
@@ -41,20 +46,96 @@ export class TeaListContainer extends React.Component {
 
       const revSortOrder = [...newSortOrder].reverse();
 
-      currentState = {
+      currentSortState = {
         columnName: columnName,
         asc: newSortOrder,
         desc: revSortOrder
       };
     }
     this.setState({
       sortedIDs: newSortOrder,
-      memoizedIDs: currentState,
+      memoizedIDs: currentSortState,
       sortColumn: columnName,
       sortOrder: sortOrder
     });
   };
 
+  filterChangeHandler = event => {
+    const name = event.target.name;
+    const value = event.target.value;
+
+    this.setState({
+      ...this.state.formControls,
```

Comment:
This will cause everything inside `formControls` to be spread onto the top level of the state, which is probably not what you want. Remove this line.

Reply:
yup. i saw that.